src/Stratis.Bitcoin.IntegrationTests/SegWitTests.cs

```diff
@@ -30,44 +28,66 @@ public void TestSegwitActivation()
                 coreNode.ConfigParameters.AddOrReplace("printtoconsole", "0");
                 coreNode.Start();
 
-                CoreNode node1 = builder.CreateStratisPowNode(true, fullNodeBuilder =>
+                CoreNode stratisNode = builder.CreateStratisPowNode(true, fullNodeBuilder =>
                 {
                     fullNodeBuilder
                         .UseConsensus()
                         .UseBlockStore()
+                        .UseWallet()
                         .UseMempool()
-                        .UseBlockNotification()
-                        .UseTransactionNotification()
                         .AddMining()
-                        .UseWallet()                        
-                        .UseApi()
                         .AddRPC();
                 });
 
-                WalletManager wm1 = node1.FullNode.NodeService<IWalletManager>() as WalletManager;
-                wm1.CreateWallet("Test1", "alice1");
-
-                RPCClient rpc1 = node1.CreateRPCClient();
+                RPCClient stratisNodeRpc = stratisNode.CreateRPCClient();
                 RPCClient coreRpc = coreNode.CreateRPCClient();
 
-                coreRpc.AddNode(node1.Endpoint, false);
-                rpc1.AddNode(coreNode.Endpoint, false);
+                coreRpc.AddNode(stratisNode.Endpoint, false);
+                stratisNodeRpc.AddNode(coreNode.Endpoint, false);
 
-                coreRpc.Generate(450);
+                // core does not mine segwit blocks by default so we disable the segwit auto activation on regtest
```

Comment:
Perhaps it is more accurate to say that core (in version 0.15.1) only mines segwit blocks above a certain height on regtest, through the activation window logic being applied?

Reply:
yes correct will update the comment.